sdk/src/main/java/com/microsoft/did/sdk/util/ReceiptUtil.kt

```
@@ -23,4 +44,15 @@ fun createReceipt(entityDid: String, entityName: String, receiptAction: ReceiptA
         entityIdentifier = entityDid,
         entityName = entityName
     )
+}
+
+/**
+ * Get receipts by verifiable credential id from the database.
+ */
+private suspend fun saveReceipt(receipt: Receipt, vchRepository: VerifiableCredentialHolderRepository): Result<Unit> {
+    return try {
+        Result.Success(vchRepository.insert(receipt))
+    } catch (exception: Exception) {
+        Result.Failure(RepositoryException("Unable to insert receipt in repository.", exception))
+    }
```

I think you copied this code from somewhere, but I don't see how it's doing anything useful. It does the same thing that runResultTry already does for us just a level above (twice in this case). Also we shouldn't catch all Exceptions (except at the API boundary which we do with runResultTry)

---

Removed this method as it doesn't serve any purpose.